lookout/style/format/code_generator.py

```diff
@@ -67,14 +71,19 @@ def apply_predicted_y(self, vnodes: Sequence[VirtualNode],
         :param vnodes_y: Sequence of the labeled `VirtualNode`-s corresponding to labeled samples.\
                          Should be ordered by start position value.
         :param y_pred: The model predictions for `vnodes_y` `VirtualNode`-s.
+        :param rule_winners: Sequence of applied rules to generate y_pred.
+        :param rules: Rules that were used for prediction.
         :return: The list of VirtualNode-s with adjusted labels. "y_old" attribute of each node \
                  contains the previous label.
         """
         result = []
-        for vnode, y_new in self._iterate_vnodes(vnodes, vnodes_y, y_pred):
+        for vnode, y_new, rule_winner in \
+                self._iterate_vnodes(vnodes, vnodes_y, y_pred, rule_winners):
             vnode = vnode.copy()
             if y_new != vnode.y:
                 vnode.y, vnode.y_old = y_new, vnode.y
+                if rule_winner:
+                    vnode.applied_rule = rules.rules[rule_winner]
```

Comment:
`else: vnode.applied_rule = None`?

Reply:
I use `hasattr()` to check if some rule were applied. If you have some arguments against I can change it.